android/app/src/main/java/com/zulipmobile/ShareContent.java

```diff
@@ -0,0 +1,93 @@
+package com.zulipmobile;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.core.content.FileProvider;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Allows Sending Content from zulip to other apps.
+ */
+
+public class ShareContent extends ReactContextBaseJavaModule {
+    private String TAG = "ShareContent";
+
+    public ShareContent(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public String getName() {
+        return "ShareContent";
+    }
+
+    @ReactMethod
+    public void shareContent(@Nullable ReadableMap source) {
```

Comment:
Can you change this to explicitly take a `url` and `auth` with the correct types? That will make this function significantly more type-safe.

Reply:
I started a CZO thread for this [here](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/logging.20to.20sentry.20from.20native.20android.20code) :)

Reply:
I have changed the signature now it is `shareFile(@Nullable String authHeader, String resourceUrl)`.